### PR TITLE
refactor: move `new jsonrpc types` to shared types

### PR DESCRIPTION
**Pull request by niklasad1**
Brute-force but I will build on this in later PRs, as I need to migrate crate by crate to this.

I wasn't in the mood on writing docs for the moved code :)

### CODE REVIEW

File: ws-server/src/server.rs [hunk 1]
```diff
@@ -28,67 +28,29 @@ use futures::io::{BufReader, BufWriter};
...
 use tokio_util::compat::TokioAsyncReadCompatExt;
 
-use crate::types::{ConnectionId, Methods, RpcId, RpcSender};
-use crate::types::{JsonRpcError, JsonRpcErrorParams};
-use crate::types::{JsonRpcInvalidRequest, JsonRpcRequest, JsonRpcResponse, TwoPointZero};
-use crate::types::{JsonRpcNotification, JsonRpcNotificationParams};
+use jsonrpsee_types::jsonrpc_v2::{ConnectionId, Methods, RpcError, RpcId, RpcParams, RpcSender};
+use jsonrpsee_types::jsonrpc_v2::{JsonRpcError, JsonRpcErrorParams};
+use jsonrpsee_types::jsonrpc_v2::{JsonRpcInvalidRequest, JsonRpcRequest, JsonRpcResponse, TwoPointZero};
+use jsonrpsee_types::jsonrpc_v2::{JsonRpcNotification, JsonRpcNotificationParams};
```
Comment (dvdplm) [inline ws-server/src/server.rs:46]: Might be worth renaming the crate to just `types`

File: types/src/lib.rs [hunk 2]
```diff
@@ -8,6 +8,9 @@ extern crate alloc;
 /// JSON-RPC 2.0 specification related types.
 pub mod jsonrpc;
 
+/// JSON-RPC 2.0 specification related types v2.
+pub mod jsonrpc_v2;
```
Comment (dvdplm) [inline types/src/lib.rs:12]: Could be just `v2` I think? External users will use `jsonrpsee_types` anyway right?

File: types/src/jsonrpc_v2/error.rs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+use thiserror::Error;
+
+/// Error.
```
Comment (dvdplm) [inline types/src/jsonrpc_v2/error.rs:3]: Some docs here maybe. Missing docs for all variants.

Comment (niklasad1) [reply types/src/jsonrpc_v2/error.rs:3]: > I wasn't in the mood on writing docs for the moved code

Yes, we need docs :)

**Review by dvdplm — APPROVED** [review]
LGTM modulo `v2` renaming.

**Comment by niklasad1** [comment]
Closing this included in #253

**Pull request closed without merging**